.github/workflows/publish-mcp-server.yaml

```diff
@@ -0,0 +1,94 @@
+name: ElasticGraph MCP Server Python Package
+
+on:
+  # Manually trigger the piepline to publish to PyPI
+  workflow_dispatch:
+  pull_request:
+    paths:
+      - 'ai_tools/elasticgraph-mcp-server/**'
+
+jobs:
+  test:
+    name: Test and Lint
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ai_tools/elasticgraph-mcp-server
+
+    steps:
+    - uses: actions/checkout@v4
```

Comment:
/nit the security best practice is to pin used actions to a sha with a comment about which version it is. I think dependabot will even maintain that. Could you update these actions to use that format? See the other workflows for examples.